src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,145 +1,392 @@
+        if(ts.length >= this.size()){
+            for (int i = 0; i < this.size; i++) {
+                ts[i] = (T) this.get(i);
+            }
+            return ts;
+        }else{
+            Object[] newResizedArray = new Object[ts.length];
+            for (int i = 0; i < ts.length; i++) {
+                newResizedArray[i] = (T) this.get(i);
+            }
+        return (T[]) newResizedArray;
+        }
     }
 
     @Override
-    public boolean add(Student student) {
-        //TODO
-        return false;
+    public boolean add(T t) {
+        if (size == elements.length) {
+            resizeArray();
+        }
+        elements[size++] = t;
+        return true;
     }
 
     @Override
-    public boolean remove(Object o) {
-        //TODO
-        return false;
+    public boolean remove(Object t) {
+        if (!this.contains(t)) {
+            return false;
+        }
+        Object[] elements = decreasedArray(size);
+        int indexOfDeletedObject = this.indexOf(t);
+        copyArray(indexOfDeletedObject, elements);
+        return true;
     }
 
     @Override
     public void clear() {
-        //TODO
+        for (int i = 0; i < size; i++)
+            elements[i] = null;
+        size = 0;
     }
 
     @Override
-    public Student get(int i) {
-        //TODO
-        return null;
+    public T get(int i) {
+        if (i >= 0 && i <= size) {
+            return (T) elements[i];
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
-    public Student set(int i, Student student) {
-        //TODO
-        return null;
+    public T set(int i, T t) {
+        if (i >= 0 && i < size) {
+            return (T) (elements[i] = t);
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
     }
 
     @Override
-    public void add(int i, Student student) {
-        //TODO
+    public void add(int i, T t) {
+        if (i < 0 && i > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (size == elements.length) {
+            resizeArray();
+        }
+        for (int start = size; start > i; start--) {
+            elements[start] = elements[start - 1];
+        }
+        elements[i] = t;
     }
 
     @Override
-    public Student remove(int i) {
-        //TODO
-        return null;
+    public T remove(int i) {
+        if (i < 0 || i >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        final Object elementToDelete = elements[i];
+        Object[] elements = decreasedArray(size);
+        copyArray(i, elements);
+        return (T) elementToDelete;
     }
 
     @Override
-    public int indexOf(Object o) {
-        //TODO
-        return 0;
+    public int indexOf(Object t) {
+        if (Objects.isNull(t)) {
+            for (int i = 0; i < size; i++) {
+                if (null == elements[i]) {
+                    return i;
+                }
+            }
+        }
```

Comment:
try to do it in one loop

Reply:
The same problem as above I need two "if" to check "null"

Reply:
Did it.